File: 2021/05/01/main.py
class WordFilter:

    def __init__(self, words: List[str]):
        self.mp = {}
        for i in range(len(words)):
            w = words[i]
            for j in range(1, len(w) + 1):
                for k in range(1, len(w) + 1):
                    st = w[0:j]+"#"+w[k-len(w):]
                    self.mp[st] = i

    def f(self, prefix: str, suffix: str) -> int:
        st = prefix + "#" + suffix
        return self.mp[st] if st in self.mp else -1
        

# Your WordFilter object will be instantiated and called as such:
# obj = WordFilter(words)
# param_1 = obj.f(prefix,suffix)

File: 2021/05/01/main.go
package main

type WordFilter struct {
	mp map[string]int
}

func Constructor(words []string) WordFilter {
	m := make(map[string]int)
	for i := 0; i < len(words); i++ {
		w := words[i]
		for j := 1; j <= len(w); j++ {
			for k := 1; k <= len(w); k++ {
				m[w[:j]+"#"+w[len(w)-k:]] = i
			}
		}
	}
	return WordFilter{mp: m}
}

func (this *WordFilter) F(prefix string, suffix string) int {
	p := prefix + "#" + suffix
	if index, ok := this.mp[p]; ok {
		return index
	}
	return -1
}

/**
 * Your WordFilter object will be instantiated and called as such:
 * obj := Constructor(words);
 * param_1 := obj.F(prefix,suffix);
 */
